client/db/bolt/db.go

```diff
@@ -247,17 +249,16 @@ func (db *BoltDB) Accounts() ([]*dexdb.AccountInfo, error) {
 }
 
 // Account gets the AccountInfo associated with the specified DEX address.
-func (db *BoltDB) Account(url string) (*dexdb.AccountInfo, error) {
+func (db *BoltDB) Account(accountID account.AccountID) (*dexdb.AccountInfo, error) {
 	var acctInfo *dexdb.AccountInfo
-	acctKey := []byte(url)
 	return acctInfo, db.acctsView(func(accts *bbolt.Bucket) error {
-		acct := accts.Bucket(acctKey)
+		acct := accts.Bucket(accountInfoKey(accountID))
```

Comment:
It seems the motivation for the change of key here is so that you can keep disabled accounts in the same bucket as active accounts. That kinda works, but have you considered instead moving disabled accounts to a different bucket altogether. You would then just returning the accounts from the active bucket from `db.Accounts`. `DisableAccount` would then move the account from the active accounts bucket to the disabled accounts bucket. Active accounts bucket can remain keyed by `host`, which is consistent with our `Core` implementation, and disabled accounts bucket can be keyed by `EncKey` (just as unique as account ID, but doesn't require decrypting the private key to generate).

Reply:
The current solution is a culmination of conversations: https://matrix.to/#/!EzTSRQITaqHuFBDFhM:decred.org/$jni3cKrEBHgGZydTyIg8s4OptEHBbdShniyHnzASAaw?via=decred.org&via=matrix.org&via=planetdecred.org and https://github.com/decred/dcrdex/pull/519#discussion_r452497210

Now that I'm intensely familiar with how all of this works I think I like your solution of creating a disabled accounts bucket with `EncKey` as key. That solves multiple issues. 

1. The data within the accounts bucket will reflect accurately the data within c.conns with host remaining as key.
2. There is not a chance that duplicate data per host will exist within the accounts bucket if host remains as key.
3. The data within the accounts bucket can be considered pristine as anything disabled is in an entirely different bucket.

The only apprehension I have is from my background of the relational database philosophy of "never destroy data" just create a status column to tell you if the data is active or not. Buckets however don't seem to offer selecting based on criteria like `status != "disabled"` so I believe the disabled accounts bucket is cleaner.

Now, onto a big revert! :)